### DISCUSSION TITLE
Vehicle Width and Height

### DISCUSSION THREAD

**Discussion by mertbakir** [valhalla/valhalla#3295] [Q&A] [answered]
I am not sure if this is correct or not so I wanted to ask:
Is this how we pass width and height of a vehicle or do I need to use a key called "vehicle". This is not clear in the [docs](https://valhalla.readthedocs.io/en/latest/api/turn-by-turn/api-reference/), IMO.

```json
"costing": "truck",
            "truck": {"length": 4, "weight": 16, "width": 2, "height": 5},
            "costing_options": {
                "truck": {"gate_penalty": 600}
            },
```

**Comment by kevinkreiser** [comment, marked as answer]
should be more like:

```json
"costing": "truck",
            "costing_options": {
                "truck": {"gate_penalty": 600, "length": 4, "weight": 16, "width": 2, "height": 5}
            },
```

all the options go inside of the particular costings costing_options section. i know its confusing but we basically did this for the sake of multimodal routing.

**Reply by mertbakir** [reply]
Thanks, got it.

**Reply by answerquest** [reply]
hi, can these costing options be included in sources-to-targets api allso?

**Reply by chrstnbwnkl** [reply]
Yes they can! The costing is available for pretty much all end points (truck dimensions are even considered in locate requests of IIRC)
